src/transxchange/read.rs

```diff
@@ -92,13 +92,14 @@ where
 }
 
 /// Read TransXChange format into a Navitia Transit Model
-pub fn read<P>(transxchange_path: P, naptan_path: P) -> Result<Model>
+pub fn read<P>(transxchange_path: P, naptan_path: P, prefix: Option<String>) -> Result<Model>
 where
     P: AsRef<Path>,
 {
     let mut collections = Collections::default();
     naptan::read_naptan(naptan_path, &mut collections)?;
     read_transxchange_archive(transxchange_path, &mut collections)?;
+    prefix.map(|p| collections.add_prefix(p.as_str()));
```

Comment:
generally map is used to return non unit `()` value
you can do 
```rust
if let Some(prefix) = prefix {
    collections.add_prefix(p.as_str());
}
```

Reply:
I knew I was doing something weird and yet my brain didn't provide the easy answer to that... Thank you. Fixed in 2250daeb61de84a7e15340e610c9b59db8220b6e.